src/components/shared/RightSidebar/RelatedPostsCMD.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+
+import {
+  Command,
+  CommandDialog,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  CommandSeparator,
+  CommandShortcut,
+} from "@/components/ui/command";
+
+interface RelatedPostsCMDProps {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  allOtherPosts?: { title: string; id: string }[];
+  addRelatedPost: (postToAdd: string) => void;
+}
+
+const RelatedPostsCMD: React.FC<RelatedPostsCMDProps> = ({
+  open,
+  setOpen,
+  allOtherPosts,
+  addRelatedPost,
+}) => {
+  return (
+    <CommandDialog open={open} onOpenChange={setOpen}>
+      <CommandInput
+        className="bg-myBlack-700 text-myWhite-100"
+        placeholder="Search for a post to add it as related..."
+      />
+      <CommandList className="customScroll bg-myBlack-900">
+        <CommandEmpty>No results found.</CommandEmpty>
+        <CommandGroup heading="Other Posts">
+          {allOtherPosts?.map((post) => {
+            return (
```

Comment:
We could filter out posts that have already been selected as related here.

We could pass in the currently related posts:
```js
// RelatedPosts.tsx
  <RelatedPostsCMD
        open={open}
        setOpen={setOpen}
        currentRelatedPosts={relatedPosts} // pass these in
        allOtherPosts={allOtherPosts}
        addRelatedPost={addRelatedPost}
      />
```


Then here, in our `RelatedPostsCMD.tsx` we can simply return early when we find that this post's id is already in our list:

```js
 <CommandList className="customScroll bg-myBlack-900">
        <CommandEmpty>No results found.</CommandEmpty>
        <CommandGroup heading="Other Posts">
          {allOtherPosts?.map((post) => {
          
// if we already have this in our currently related posts, don't render it as an option
            if (currentRelatedPosts?.find((p) => p.id === post.id)) return;
            return (
              <CommandItem
                key={post.id}
                onClick={() => {
                  addRelatedPost(post.id);
                  setOpen(false);
                }}
              >
  ```